test/install_test.go

```diff
@@ -261,3 +300,85 @@ func TestCheckProxy(t *testing.T) {
 		t.Fatal(err.Error())
 	}
 }
+
+func TestLogs(t *testing.T) {
+	controllerRegex := regexp.MustCompile("level=(panic|fatal|error|warn)")
+	proxyRegex := regexp.MustCompile(fmt.Sprintf("%s (ERR|WARN)", proxyContainer))
+
+	for deploy, spec := range linkerdDeployReplicas {
+		deploy := strings.TrimPrefix(deploy, "linkerd-")
+		containers := append(spec.containers, proxyContainer)
+
+		for _, container := range containers {
+			regex := controllerRegex
+			if container == proxyContainer {
+				regex = proxyRegex
+			}
+
+			outputStream, err := TestHelper.LinkerdRunStream(
+				"logs", "--no-color",
+				"--control-plane-component", deploy,
+				"--container", container,
+			)
+			if err != nil {
+				t.Fatalf("Error running command:\n%s", err)
+			}
+			defer outputStream.Stop()
+			outputLines, _ := outputStream.ReadUntil(100, 10*time.Second)
+			if len(outputLines) == 0 {
+				t.Fatalf("No logs found for %s/%s", deploy, container)
+			}
+
+			for _, line := range outputLines {
+				if regex.MatchString(line) {
+
+					// check for known errors
+					known := false
+					if TestHelper.TLS() && container == proxyContainer {
+						for _, er := range knownProxyErrors {
+							if strings.HasSuffix(line, er) {
+								known = true
+								break
+							}
+						}
+					}
+
+					if !known {
+						t.Fatalf("Found error in %s/%s log: %s", deploy, container, line)
+					}
+				}
+			}
+		}
+	}
+}
+
+func TestRestarts(t *testing.T) {
+	for deploy, spec := range linkerdDeployReplicas {
+		deploy := strings.TrimPrefix(deploy, "linkerd-")
+		containers := append(spec.containers, proxyContainer, initContainer)
+
+		for _, container := range containers {
+			selector := fmt.Sprintf("linkerd.io/control-plane-component=%s", deploy)
+			containerStatus := "containerStatuses"
+			if container == initContainer {
+				containerStatus = "initContainerStatuses"
+			}
+			output := fmt.Sprintf("jsonpath='{.items[*].status.%s[?(@.name==\"%s\")].restartCount}'", containerStatus, container)
```

Comment:
Ah interesting. Instead of shelling out to Kubernetes for the restart check, I was originally envisioning that we'd add a new test helper to do this programmatically, in `testutil/kubernetes_helper.go`, similar to what `CheckPods` is doing. In fact, how would you feel about just updating the `CheckPods` to also validate that none of the containers have restarted, and then calling that here after the rest of the tests in this file have run? In might take a small refactor, but I think it would be useful generally, and could be handy in our other tests that call `CheckPods`.